ipc/provider/src/checkpoint.rs

```diff
@@ -89,20 +92,62 @@ impl<T: BottomUpCheckpointRelayer + Send + Sync + 'static> BottomUpCheckpointMan
     }
 
     /// Run the bottom up checkpoint submission daemon in the background
```

Comment:
We are actually running the process in the foreground.
```suggestion
    /// Run the bottom up checkpoint submission daemon in the foreground
```